flink-runtime/src/main/java/org/apache/flink/runtime/state/metrics/LatencyTrackingStateConfig.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+
+/** Config to create latency tracking state metric. */
+public class LatencyTrackingStateConfig {
+
+    private final MetricGroup metricGroup;
+
+    private final boolean enabled;
+    private final int sampleInterval;
+    private final long slidingWindow;
+
+    LatencyTrackingStateConfig(
+            MetricGroup metricGroup, boolean enabled, int sampleInterval, long slidingWindow) {
+        this.metricGroup = metricGroup;
+        this.enabled = enabled;
+        this.sampleInterval = sampleInterval;
+        this.slidingWindow = slidingWindow;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public MetricGroup getMetricGroup() {
+        return metricGroup;
+    }
+
+    public long getSlidingWindow() {
+        return slidingWindow;
+    }
+
+    public int getSampleInterval() {
+        return sampleInterval;
+    }
+
+    public static LatencyTrackingStateConfig disabled() {
+        return newBuilder().setEnabled(false).build();
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        private boolean enabled = StateBackendOptions.LATENCY_TRACK_ENABLED.defaultValue();
+        private int sampleInterval =
+                StateBackendOptions.LATENCY_TRACK_SAMPLE_INTERVAL.defaultValue();
+        private long slidingWindow =
+                StateBackendOptions.LATENCY_TRACK_SLIDING_WINDOW.defaultValue();
+        private MetricGroup metricGroup;
+
+        public Builder setEnabled(boolean enabled) {
+            this.enabled = enabled;
+            return this;
+        }
+
+        public Builder setSampleInterval(int sampleInterval) {
+            this.sampleInterval = sampleInterval;
+            return this;
+        }
+
+        public Builder setSlidingWindow(long slidingWindow) {
+            this.slidingWindow = slidingWindow;
+            return this;
+        }
+
+        public Builder setMetricGroup(MetricGroup metricGroup) {
+            this.metricGroup = metricGroup;
+            return this;
+        }
+
+        public Builder configure(ReadableConfig config) {
+            this.setEnabled(config.get(StateBackendOptions.LATENCY_TRACK_ENABLED))
+                    .setSampleInterval(
+                            config.get(StateBackendOptions.LATENCY_TRACK_SAMPLE_INTERVAL))
+                    .setSlidingWindow(config.get(StateBackendOptions.LATENCY_TRACK_SLIDING_WINDOW));
+            return this;
+        }
+
+        public LatencyTrackingStateConfig build() {
+            if (enabled) {
+                Preconditions.checkNotNull(
+                        metricGroup, "Metric group cannot be null if latency tracking is enabled.");
+                Preconditions.checkArgument(sampleInterval >= 1);
```

Comment:
Why `0` is not allowed? I.e., track **every** state access?

Reply:
I think current description of **'interval'** would be misleading as its original meaning should be tracking every **n** state access. I'll fix them.

Reply:
After reading some more references, I think `sample interval` should be okay as I actually select [systematic sampling](https://en.wikipedia.org/wiki/Systematic_sampling) in which the `sample interval` just have such definition.
If we select `record-1` as the first one, and choose 100 as the interval. The next record to select is `record-101`.

Reply:
I personally don't see any reason why sampling interval can not be zero (the linked article doesn't discuss it), and therefore those descriptions would be misleading (I'm also fine with disallowing `0`).

Reply:
You can read some more descriptions on [systematic sampling](https://www.questionpro.com/blog/systematic-sampling/) in which the sample interval is defined as:
~~~
Systematic Sampling Formula for interval (i) = N/n = 5000/500 = 10
~~~

Reply:
Makes sense.